cmsio1.c: In function 'AdjustEndianessArray16':
cmsio1.c:118: warning: implicit declaration of function 'swab'
testcms.c: In function 'CheckSwab':
testcms.c:113: warning: implicit declaration of function 'swab'
================================================================================
--- src/Makefile.am
+++ src/Makefile.am
@@ -6,6 +6,9 @@
 # Don't require all the GNU mandated files
 AUTOMAKE_OPTIONS = 1.7 foreign
 
+# _XOPEN_SOURCE is required for swab() but undefines strcasecmp(), so add _GNU_SOURCE as well:
+AM_CPPFLAGS = -D_XOPEN_SOURCE -D_GNU_SOURCE
+
 includedir = ${prefix}/include
 
 # Shared libraries built in this directory
--- src/cmsio1.c
+++ src/cmsio1.c
@@ -24,6 +24,7 @@
 
 
 #include "lcms.h"
+#include <unistd.h>
 
 // ----------------------------------------------------------------- Tag Serialization
 
--- testbed/Makefile.am
+++ testbed/Makefile.am
@@ -6,6 +6,9 @@
 # Don't require all the GNU mandated files
 AUTOMAKE_OPTIONS = 1.7 foreign
 
+# Required for swab()
+AM_CPPFLAGS = -D_XOPEN_SOURCE
+
 INCLUDES	=  -I$(top_builddir)/include -I$(top_srcdir)/include
 
 check_PROGRAMS = testcms
--- testbed/testcms.c
+++ testbed/testcms.c
@@ -36,6 +36,7 @@
 #include <time.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #ifndef NON_WINDOWS
 #include <icm.h>
